File: medplum-java/medplum-model/src/main/java/com/fhir/RiskAssessment_Prediction.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.Double;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableRiskAssessment_Prediction.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface RiskAssessment_Prediction {
  Optional<decimal> relativeRisk();

  Optional<List<Extension>> extension();

  Optional<Period> whenPeriod();

  Optional<String> rationale();

  Optional<Range> probabilityRange();

  Optional<CodeableConcept> qualitativeRisk();

  Optional<Range> whenRange();

  Optional<String> id();

  Optional<List<Extension>> modifierExtension();

  Optional<CodeableConcept> outcome();

  Optional<Double> probabilityDecimal();

  static ImmutableRiskAssessment_Prediction.Builder builder() {
    return ImmutableRiskAssessment_Prediction.builder();
  }
}

File: medplum-typescript/src/model/fhir/RiskAssessment_Prediction.ts
import { CodeableConcept } from "./CodeableConcept";
import { decimal } from "./decimal";
import { Extension } from "./Extension";
import { Period } from "./Period";
import { Range } from "./Range";

export interface RiskAssessment_Prediction {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    outcome: CodeableConcept | null | undefined;
    probabilityDecimal: number | null | undefined;
    probabilityRange: Range | null | undefined;
    qualitativeRisk: CodeableConcept | null | undefined;
    relativeRisk: decimal | null | undefined;
    whenPeriod: Period | null | undefined;
    whenRange: Range | null | undefined;
    rationale: string | null | undefined;
}
